module/VuFind/src/VuFind/Content/Covers/Bokinfo.php

```diff
@@ -0,0 +1,204 @@
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Jacob Sandin <jacob@imcode.com>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     Wiki
+ * https://vufind.org/wiki/development:plugins:content_provider_components#covers
+ */
+namespace VuFind\Content\Covers;
+
+use SimpleXMLElement;
+
+/**
+ * Summon cover content loader.
+ *
+ * @category VuFind
+ * @package  Content
+ * @author   Demian Katz <demian.katz@villanova.edu>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development Wiki
+ */
+class Bokinfo extends \VuFind\Content\AbstractCover implements
+    \Laminas\Log\LoggerAwareInterface,
+    \VuFindHttp\HttpServiceAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait;
+    use \VuFindHttp\HttpServiceAwareTrait;
+
+    /**
+     * Constructor
+     */
+    public function __construct()
+    {
+        $this->supportsIsbn = $this->cacheAllowed = true;
+    }
+
+    /**
+     * Get image URL for a particular API key and set of IDs (or false if invalid).
+     *
+     * @param string $key  API key
+     * @param string $size Size of image to load (small/medium/large)
+     * @param array  $ids  Associative array of identifiers (keys may include 'isbn'
+     * pointing to an ISBN object and 'issn' pointing to a string)
+     *
+     * @return string|bool
+     */
+    public function getUrl($key, $size, $ids)
+    {
+        if (!isset($ids['isbn'])) {
+            return false;
+        }
+        if (!isset($key)) {
+            return false;
+        }
+
+        $isbn = $ids['isbn']->get13();
+        try {
+            $client = $this->createHttpClient(
+                "https://api.bokinfo.se/book/get/$isbn"
+            );
+
+            $client->getRequest()->getHeaders()
+                  ->addHeaderLine("Ocp-Apim-Subscription-Key", $key);
+
+            try {
+                $resp= $client->send();
+                $body = $resp->getBody();
+                $url = $this->getImageUrl($body);
+            } catch (\Exception $e) {
+            }
+
+            if ($this->testUrlFunction($url)) {
+                return "$url";
+            }
+        } catch (\Throwable $ex) {
+            return false;
+        } catch (\Exception $ex) {
+            return false;
+        } catch (\RuntimeException $ex) {
+            return false;
+        }
+
+        return false;
+    }
+
+    /**
+     * Return a HTTP Client object
+     *
+     * @param string $url API Url
+     *
+     * @return HttpClient Http Client
+     */
+    protected function createHttpClient($url)
+    {
+        $client = $this->httpService->createClient($url);
+
+        if (isset($this->config['Http']['ssl_verify_peer_name'])
```

Comment:
Where is `$this->config` supposed to be coming from? It doesn't seem to be defined or set anywhere, which is why PHPStan is complaining. Let me know if you need help with configuration injection.

Is there a strong reason to put all this configuration logic here? If the ssl_verify_peer_name option is set, does it need to be handled differently by Bokinfo than by everything else?

Reply:
I changed so that it just verify and set timeout to 30. It was copied code from KohaRest I think. The debugger did not complain so I thought nothing of it until now when tests failed.

Reply:
That's the beauty of static analysis! :-) The code worked because it was checking if the value was set before doing anything, so it wasn't doing anything... but PHPStan was smart enough to figure out that the value could NEVER be set, so it gave us a warning.